### DISCUSSION TITLE
Equivalent classes in bc-fips-1.0.2.4.jar

### DISCUSSION THREAD

**Discussion by smartycardpants** [bcgit/bc-java#1592] [Q&A] [answered]
I was using bc-fips-1.0.2.jar until becoming aware of CVEs associated with that jar. I was using the following classes:

BlockCipher
AESEngine
CMac
KeyParameter

Now those classes are invisible in bc-fips-1.0.2.4.jar.

What are the equivalent classes in bc-fips-1.0.2.4.jar ?

**Comment by smartycardpants** [comment]
I found these somewhat equivalent classes in bc-fips-1.0.2.3.jar:

import org.bouncycastle.crypto.SymmetricKey;
import org.bouncycastle.crypto.SymmetricSecretKey;
import org.bouncycastle.crypto.fips.FipsAES;
import org.bouncycastle.crypto.fips.FipsMACOperatorFactory;
import org.bouncycastle.crypto.fips.FipsOutputMACCalculator;

Will these classes be carried forward "as is" to bc-fips-1.0.2.5.jar?

**Reply by dghgit** [reply, marked as answer]
Yes.
